tests/test_pymc.py

```diff
@@ -127,18 +127,34 @@ def test_det(backend, gradient_backend):
     assert trace.posterior.b.shape[-1] == 2
 
 
+@parameterize_backends
+def test_non_identifier_names(backend, gradient_backend):
```

Comment:
What are bon identifier names? What was failing?

Reply:
The shared variables are typically hidden in the pytensor function, and that uses positional args.
And the strange pymc point functions also assume things are in the right order (which has bitten me more than once...)
I called it non-identifier because this happens more or less if `name.isidentifier()` is false.

Reply:
I guess we just don't use it much internally because it would definitely fail there as well: https://github.com/pymc-devs/pymc/blob/main/pymc%2Fmodel%2Fcore.py#L2201

Anyway it was just curiosity, changes look good

Reply:
It does work:

```
import pymc as pm

with pm.Model() as model:
    pm.Normal("a/b")

point = model.initial_point()
func = model.compile_logp(mode="JAX")
func(point)
```

This call pytensor_func(**point), but that converts it to positional arguments for the jax function.

Reply:
Ah it's a jax limitation, not a Python one??

Reply:
Where do we convert it to positional arguments? Inside PyTensor? Because here it's just unpacking the dict: https://github.com/pymc-devs/pymc/blob/main/pymc%2Fpytensorf.py#L617-L624

Reply:
Anyway I thought the problem is you cant unpack non identifiers in Python

Reply:
Nope, no limitation anywhere :-)

It is perfectly legal in python (and I think also in jax) to have a function

```
def foo(**kwargs):
    pass
```
And then pass it something like `{"a/b": 1}`. As long as foo takes the kwargs as `**kwargs` that's allowed.

The problem is that the jax function we generate in dispatch generates something like
```
def jax_dispatch(a_b):
    pass
```
and we can't call that function with `foo(**{"a/b": 1})`, because the name of the variable and the key in the dictionary don't match.

Reply:
Hmm right I didn't expect those internal variables to ever be called with kwargs

Reply:
Or for the names to even respect those from the original graph